client/app/components/DialogWrapper.jsx

```
@@ -0,0 +1,172 @@
+         this.props.dialog.close({ success: true }); // or dismiss();
+      });
+    }
+
+    render() {
+      const { dialog } = this.props;
+        return (
+          <Modal {...dialog.props} onOk={() => this.customOkHandler()}>
+        );
+    }
+
+*/
+
+export const DialogPropType = PropTypes.shape({
+  props: PropTypes.shape({
+    visible: PropTypes.bool,
+    onOk: PropTypes.func,
+    onCancel: PropTypes.func,
+    afterClose: PropTypes.func,
+  }).isRequired,
+  close: PropTypes.func.isRequired,
+  dismiss: PropTypes.func.isRequired,
+});
+
+function openDialog(DialogComponent, props) {
+  const dialog = {
+    props: {
+      visible: true,
+      onOk: () => {},
+      onCancel: () => {},
+      afterClose: () => {},
+    },
+    close: () => {},
+    dismiss: () => {},
+  };
+
+  const dialogResult = {
+    resolve: () => {},
+    reject: () => {},
+  };
+
+  const container = document.createElement('div');
+  document.body.appendChild(container);
+
+  function render() {
+    ReactDOM.render(<DialogComponent {...props} dialog={dialog} />, container);
+  }
+
+  function destroyDialog() {
+    // Allow calling chain to roll up, and then destroy component
+    setTimeout(() => {
+      ReactDOM.unmountComponentAtNode(container);
+      document.body.removeChild(container);
+    }, 10);
+  }
+
+  function closeDialog(result) {
+    dialogResult.resolve(result);
+    dialog.props.visible = false;
+    render();
+  }
+
+  function dismissDialog(reason) {
+    dialogResult.reject(reason);
+    dialog.props.visible = false;
+    render();
+  }
+
+  dialog.props.onOk = closeDialog;
+  dialog.props.onCancel = dismissDialog;
+  dialog.props.afterClose = destroyDialog;
+  dialog.close = closeDialog;
+  dialog.dismiss = dismissDialog;
+
+  const result = {
+    close: closeDialog,
+    dismiss: dismissDialog,
+    result: new Promise((resolve, reject) => {
+      dialogResult.resolve = resolve;
+      dialogResult.reject = reject;
+    }),
```

I believe there would be an "unhandled rejection" exception if the other end doesn't attach a `result.catch()`.
But you can add an empty `.finally(() => {})` here to suppress the error.

---

@ranbena  `.finally(() => {})` doesn't help :disappointed: and there is no way to suppress this error in browsers. Some libraries like Bluebird have dedicated API to suppress unhandled rejection error on particular promise instance. So I added an option to override support library used by this wrapper, and also added support for known features (ATM only Bluebird). Also I set up this wrapper to use AngularJS promise (`$q`) which does not show this error. When Angular will be finally removed we can return to this (probably will start to use Bluebird, or just ignore this kind of errors).